units/patch_intent.go

```diff
@@ -491,55 +493,44 @@ func (j *patchIntentProcessor) getPreviousPatchDefinition(project *model.Project
 	return res, nil
 }
 
-func (j *patchIntentProcessor) getPreviousPatchFailures(project *model.Project) ([]patch.VariantTasks, error) {
-	previousPatch, err := patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), project.Identifier))
+func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, displayTasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, []patch.DisplayTask, error) {
+	failedTasks, err := task.FindAll(db.Query(task.FailedExecutionTasksByVersionAndBV(version, vt.Variant)))
 	if err != nil {
-		return nil, errors.Wrap(err, "error querying for most recent patch")
-	}
-	if previousPatch == nil {
-		return nil, errors.Errorf("no previous patch available")
+		return nil, nil, errors.Wrap(err, "error querying for failed tasks from previous patch")
 	}
 
-	var res []patch.VariantTasks
-
-	if previousPatch.Status == evergreen.PatchFailed {
-		for _, vt := range previousPatch.VariantsTasks {
-			failedTasks, err := task.FindAll(db.Query(task.FailedTasksByVersionAndBV(previousPatch.Version, vt.Variant)))
-			if err != nil {
-				return nil, errors.Wrap(err, "error querying for failed tasks from previous patch")
-			}
-			tasksInProjectVariant := project.FindTasksForVariant(vt.Variant)
-
-			failedTaskNames := make([]string, 0, len(failedTasks))
-			for _, failedTask := range failedTasks {
-				if !failedTask.DisplayOnly && !failedTask.IsPartOfDisplay() {
-					failedTaskNames = append(failedTaskNames, failedTask.DisplayName)
-				}
-			}
+	failedTaskNames := make([]string, 0, len(failedTasks))
+	for _, failedTask := range failedTasks {
+		failedTaskNames = append(failedTaskNames, failedTask.DisplayName)
+	}
+	// I want the subset of vt.tasks that exist in tasksForVariant
```

Comment:
I think this was my comment so womp but can you change this to "We" and add a period?